src/components/TodoComponents/TodoList.js

```diff
@@ -1,2 +1,302 @@
+                this.minutesDom = document.querySelector('#minutes');
+                this.secondsDom = document.querySelector('#seconds');
+                this.fillerDom = document.querySelector('#filler');
+                this.interval = setInterval(function () {
+                    self.intervalCallback.apply(self);
+                }, 1000);
+                document.querySelector('#work').onclick = function () {
+                    self.startWork.apply(self);
+                };
+                document.querySelector('#shortBreak').onclick = function () {
+                    self.startShortBreak.apply(self);
+                };
+                document.querySelector('#longBreak').onclick = function () {
+                    self.startLongBreak.apply(self);
+                };
+                document.querySelector('#stop').onclick = function () {
+                    self.stopTimer.apply(self);
+                };
+            },
+            resetVariables: function (mins, secs, started) {
+                this.minutes = mins;
+                this.seconds = secs;
+                this.started = started;
+                this.fillerIncrement = 200 / (this.minutes * 60);
+                this.fillerHeight = 0;
+            },
+            startWork: function () {
+                this.resetVariables(25, 0, true);
+            },
+            startShortBreak: function () {
+                this.resetVariables(5, 0, true);
+            },
+            startLongBreak: function () {
+                this.resetVariables(15, 0, true);
+            },
+            stopTimer: function () {
+                this.resetVariables(25, 0, false);
+                this.updateDom();
+            },
+            toDoubleDigit: function (num) {
+                if (num < 10) {
+                    return "0" + parseInt(num, 10);
+                }
+                return num;
+            },
+            updateDom: function () {
+                this.minutesDom.innerHTML = this.toDoubleDigit(this.minutes);
+                this.secondsDom.innerHTML = this.toDoubleDigit(this.seconds);
+                this.fillerHeight = this.fillerHeight + this.fillerIncrement;
+                this.fillerDom.style.height = this.fillerHeight + 'px';
+            },
+            intervalCallback: function () {
+                if (!this.started) return false;
+                if (this.seconds == 0) {
+                    if (this.minutes == 0) {
+                        this.timerComplete();
+                        return;
+                    }
+                    this.seconds = 59;
+                    this.minutes--;
+                } else {
+                    this.seconds--;
+                }
+                this.updateDom();
+            },
+            timerComplete: function () {
+                this.started = false;
+                this.fillerHeight = 0;
+            }
+        };
+        window.onload = function () {
```

Comment:
we will be learning lifecycle methods next week so you will know the react way of running a function onload soon. But, if all of your variables existed in state, you would not have needed to have an onload.